File: qt/gen_qsplashscreen.cpp
// +build ignore

#include <QColor>
#include <QEvent>
#include <QMetaObject>
#include <QMouseEvent>
#include <QObject>
#include <QPaintDevice>
#include <QPainter>
#include <QPixmap>
#include <QScreen>
#include <QSplashScreen>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QWidget>
#include <qsplashscreen.h>
#include "gen_qsplashscreen.h"
class MiqtVirtualQSplashScreen : public virtual QSplashScreen {
public:
MiqtVirtualQSplashScreen(): QSplashScreen() {};
MiqtVirtualQSplashScreen(QScreen* screen): QSplashScreen(screen) {};
MiqtVirtualQSplashScreen(const QPixmap& pixmap): QSplashScreen(pixmap) {};
MiqtVirtualQSplashScreen(const QPixmap& pixmap, Qt::WindowFlags f): QSplashScreen(pixmap, f) {};
MiqtVirtualQSplashScreen(QScreen* screen, const QPixmap& pixmap): QSplashScreen(screen, pixmap) {};
MiqtVirtualQSplashScreen(QScreen* screen, const QPixmap& pixmap, Qt::WindowFlags f): QSplashScreen(screen, pixmap, f) {};

virtual ~MiqtVirtualQSplashScreen() = default;
// cgo.Handle value for overwritten implementation
	intptr_t handle__MetaObject = 0;
// Subclass to allow providing a Go implementation
	virtual const QMetaObject* metaObject() const override {
if (handle__MetaObject == 0) {
return QSplashScreen::metaObject();
}
QMetaObject* callback_return_value = miqt_exec_callback_QSplashScreen_MetaObject(const_cast<MiqtVirtualQSplashScreen*>(this), handle__MetaObject);

		return callback_return_value;
	}
// Wrapper to allow calling protected method
	QMetaObject* virtualbase_MetaObject() const {

		return (QMetaObject*) QSplashScreen::metaObject();

	}
// cgo.Handle value for overwritten implementation
	intptr_t handle__Metacast = 0;
// Subclass to allow providing a Go implementation
	virtual void* qt_metacast(const char* param1) override {
if (handle__Metacast == 0) {
return QSplashScreen::qt_metacast(param1);
}
const char* sigval1 = (const char*) param1;

		void* callback_return_value = miqt_exec_callback_QSplashScreen_Metacast(this, handle__Metacast, sigval1);

		return callback_return_value;
	}
// Wrapper to allow calling protected method
	void* virtualbase_Metacast(const char* param1) {

		return QSplashScreen::qt_metacast(param1);

	}
};
QSplashScreen* QSplashScreen_new() {
return new MiqtVirtualQSplashScreen();
}
QSplashScreen* QSplashScreen_new2(QScreen* screen) {
return new MiqtVirtualQSplashScreen(screen);
}
QSplashScreen* QSplashScreen_new3(QPixmap* pixmap) {
return new MiqtVirtualQSplashScreen(*pixmap);
}
QSplashScreen* QSplashScreen_new4(QPixmap* pixmap, int f) {
return new MiqtVirtualQSplashScreen(*pixmap, static_cast<Qt::WindowFlags>(f));
}
QSplashScreen* QSplashScreen_new5(QScreen* screen, QPixmap* pixmap) {
return new MiqtVirtualQSplashScreen(screen, *pixmap);
}
QSplashScreen* QSplashScreen_new6(QScreen* screen, QPixmap* pixmap, int f) {
return new MiqtVirtualQSplashScreen(screen, *pixmap, static_cast<Qt::WindowFlags>(f));
}
void QSplashScreen_virtbase(QSplashScreen* src
, QWidget** outptr_QWidget
) {
*outptr_QWidget = static_cast<QWidget*>(src);
}
QMetaObject* QSplashScreen_MetaObject(const QSplashScreen* self) {
	return (QMetaObject*) self->metaObject();
}
void* QSplashScreen_Metacast(QSplashScreen* self, const char* param1) {
	return self->qt_metacast(param1);
}
struct miqt_string QSplashScreen_Tr(const char* s) {
	QString _ret = QSplashScreen::tr(s);
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
void QSplashScreen_SetPixmap(QSplashScreen* self, QPixmap* pixmap) {
	self->setPixmap(*pixmap);
}
QPixmap* QSplashScreen_Pixmap(const QSplashScreen* self) {
	return new QPixmap(self->pixmap());
}
void QSplashScreen_Finish(QSplashScreen* self, QWidget* w) {
	self->finish(w);
}
void QSplashScreen_Repaint(QSplashScreen* self) {
	self->repaint();
}
struct miqt_string QSplashScreen_Message(const QSplashScreen* self) {
	QString _ret = self->message();
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
void QSplashScreen_ShowMessage(QSplashScreen* self, struct miqt_string message) {
	QString message_QString = QString::fromUtf8(message.data, message.len);
	self->showMessage(message_QString);
}
void QSplashScreen_ClearMessage(QSplashScreen* self) {
	self->clearMessage();
}
void QSplashScreen_MessageChanged(QSplashScreen* self, struct miqt_string message) {
	QString message_QString = QString::fromUtf8(message.data, message.len);
	self->messageChanged(message_QString);
}
void QSplashScreen_connect_MessageChanged(QSplashScreen* self, intptr_t slot) {
	MiqtVirtualQSplashScreen::connect(self, static_cast<void (QSplashScreen::*)(const QString&)>(&QSplashScreen::messageChanged), self, [=](const QString& message) {
		const QString message_ret = message;
		// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
		QByteArray message_b = message_ret.toUtf8();
		struct miqt_string message_ms;
		message_ms.len = message_b.length();
		message_ms.data = static_cast<char*>(malloc(message_ms.len));
		memcpy(message_ms.data, message_b.data(), message_ms.len);
		struct miqt_string sigval1 = message_ms;
		miqt_exec_callback_QSplashScreen_MessageChanged(slot, sigval1);
	});
}
struct miqt_string QSplashScreen_Tr2(const char* s, const char* c) {
	QString _ret = QSplashScreen::tr(s, c);
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
struct miqt_string QSplashScreen_Tr3(const char* s, const char* c, int n) {
	QString _ret = QSplashScreen::tr(s, c, static_cast<int>(n));
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
void QSplashScreen_ShowMessage2(QSplashScreen* self, struct miqt_string message, int alignment) {
	QString message_QString = QString::fromUtf8(message.data, message.len);
	self->showMessage(message_QString, static_cast<int>(alignment));
}
void QSplashScreen_ShowMessage3(QSplashScreen* self, struct miqt_string message, int alignment, QColor* color) {
	QString message_QString = QString::fromUtf8(message.data, message.len);
	self->showMessage(message_QString, static_cast<int>(alignment), *color);
}
void QSplashScreen_override_virtual_MetaObject(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQSplashScreen*>( (QSplashScreen*)(self) )->handle__MetaObject = slot;
}
QMetaObject* QSplashScreen_virtualbase_MetaObject(const void* self) {
	return ( (const MiqtVirtualQSplashScreen*)(self) )->virtualbase_MetaObject();
}
void QSplashScreen_override_virtual_Metacast(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQSplashScreen*>( (QSplashScreen*)(self) )->handle__Metacast = slot;
}
void* QSplashScreen_virtualbase_Metacast(void* self, const char* param1) {
	return ( (MiqtVirtualQSplashScreen*)(self) )->virtualbase_Metacast(param1);
}
void QSplashScreen_Delete(QSplashScreen* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<MiqtVirtualQSplashScreen*>( self );
	} else {
		delete self;
	}
}

File: qt/gen_qsplashscreen.go
package qt

import (
	"unsafe"
)

type QSplashScreen struct {
	h          uintptr
	isSubclass bool
}

// NewQSplashScreen constructs a new QSplashScreen object.
func NewQSplashScreen() *QSplashScreen {
	g := newQSplashScreen(QSplashScreen_new())
	g.isSubclass = true
	return g
}

// NewQSplashScreen2 constructs a new QSplashScreen object.
func NewQSplashScreen2(screen *QScreen) *QSplashScreen {
	g := newQSplashScreen(QSplashScreen_new2(screen.cPointer()))
	g.isSubclass = true
	return g
}

// NewQSplashScreen3 constructs a new QSplashScreen object.
func NewQSplashScreen3(pixmap *QPixmap) *QSplashScreen {
	g := newQSplashScreen(QSplashScreen_new3(pixmap.cPointer()))
	g.isSubclass = true
	return g
}

// NewQSplashScreen4 constructs a new QSplashScreen object.
func NewQSplashScreen4(pixmap *QPixmap, f WindowType) *QSplashScreen {
	g := newQSplashScreen(QSplashScreen_new4(pixmap.cPointer(), (int)(f)))
	g.isSubclass = true
	return g
}

// NewQSplashScreen5 constructs a new QSplashScreen object.
func NewQSplashScreen5(screen *QScreen, pixmap *QPixmap) *QSplashScreen {
	g := newQSplashScreen(QSplashScreen_new5(screen.cPointer(), pixmap.cPointer()))
	g.isSubclass = true
	return g
}

// NewQSplashScreen6 constructs a new QSplashScreen object.
func NewQSplashScreen6(screen *QScreen, pixmap *QPixmap, f WindowType) *QSplashScreen {
	g := newQSplashScreen(QSplashScreen_new6(screen.cPointer(), pixmap.cPointer(), (int)(f)))
	g.isSubclass = true
	return g
}

func (this *QSplashScreen) MetaObject() *QMetaObject {
	return newQMetaObject(QSplashScreen_MetaObject(this.h))
}

func (this *QSplashScreen) Metacast(param1 string) unsafe.Pointer {
	param1_Cstring := CString(param1)
	defer free(unsafe.Pointer(param1_Cstring))
	return (unsafe.Pointer)(QSplashScreen_Metacast(this.h, param1_Cstring))
}

func QSplashScreen_Tr(s string) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	var _ms struct_miqt_string = QSplashScreen_Tr(s_Cstring)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QSplashScreen) SetPixmap(pixmap *QPixmap) {
	QSplashScreen_SetPixmap(this.h, pixmap.cPointer())
}

func (this *QSplashScreen) Pixmap() *QPixmap {
	_goptr := newQPixmap(QSplashScreen_Pixmap(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QSplashScreen) Finish(w *QWidget) {
	QSplashScreen_Finish(this.h, w.cPointer())
}

func (this *QSplashScreen) Repaint() {
	QSplashScreen_Repaint(this.h)
}

func (this *QSplashScreen) Message() string {
	var _ms struct_miqt_string = QSplashScreen_Message(this.h)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QSplashScreen) ShowMessage(message string) {
	message_ms := struct_miqt_string{}
	message_ms.data = CString(message)
	message_ms.len = size_t(len(message))
	defer free(unsafe.Pointer(message_ms.data))
	QSplashScreen_ShowMessage(this.h, message_ms)
}

func (this *QSplashScreen) ClearMessage() {
	QSplashScreen_ClearMessage(this.h)
}

func (this *QSplashScreen) MessageChanged(message string) {
	message_ms := struct_miqt_string{}
	message_ms.data = CString(message)
	message_ms.len = size_t(len(message))
	defer free(unsafe.Pointer(message_ms.data))
	QSplashScreen_MessageChanged(this.h, message_ms)
}

func (this *QSplashScreen) OnMessageChanged(slot func(message string)) {
	QSplashScreen_connect_MessageChanged(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QSplashScreen_MessageChanged
func miqt_exec_callback_QSplashScreen_MessageChanged(cb intptr_t, message struct_miqt_string) {
	gofunc, ok := cgo.Handle(cb).Value().(func(message string))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	var message_ms struct_miqt_string = message
	message_ret := GoStringN(message_ms.data, int(int64(message_ms.len)))
	free(unsafe.Pointer(message_ms.data))
	slotval1 := message_ret

	gofunc(slotval1)
}

func QSplashScreen_Tr2(s string, c string) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	c_Cstring := CString(c)
	defer free(unsafe.Pointer(c_Cstring))
	var _ms struct_miqt_string = QSplashScreen_Tr2(s_Cstring, c_Cstring)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func QSplashScreen_Tr3(s string, c string, n int) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	c_Cstring := CString(c)
	defer free(unsafe.Pointer(c_Cstring))
	var _ms struct_miqt_string = QSplashScreen_Tr3(s_Cstring, c_Cstring, (int)(n))
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QSplashScreen) ShowMessage2(message string, alignment int) {
	message_ms := struct_miqt_string{}
	message_ms.data = CString(message)
	message_ms.len = size_t(len(message))
	defer free(unsafe.Pointer(message_ms.data))
	QSplashScreen_ShowMessage2(this.h, message_ms, (int)(alignment))
}

func (this *QSplashScreen) ShowMessage3(message string, alignment int, color *QColor) {
	message_ms := struct_miqt_string{}
	message_ms.data = CString(message)
	message_ms.len = size_t(len(message))
	defer free(unsafe.Pointer(message_ms.data))
	QSplashScreen_ShowMessage3(this.h, message_ms, (int)(alignment), color.cPointer())
}

func (this *QSplashScreen) callVirtualBase_MetaObject() *QMetaObject {
	return newQMetaObject(QSplashScreen_virtualbase_MetaObject(unsafe.Pointer(this.h)))
}

func (this *QSplashScreen) OnMetaObject(slot func(super func() *QMetaObject) *QMetaObject) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QSplashScreen_override_virtual_MetaObject(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QSplashScreen_MetaObject
func miqt_exec_callback_QSplashScreen_MetaObject(self QSplashScreen, cb intptr_t) *QMetaObject {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func() *QMetaObject) *QMetaObject)
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	virtualReturn := gofunc((&QSplashScreen{h: self}).callVirtualBase_MetaObject)

	return virtualReturn.cPointer()
}

func (this *QSplashScreen) callVirtualBase_Metacast(param1 string) unsafe.Pointer {
	param1_Cstring := CString(param1)
	defer free(unsafe.Pointer(param1_Cstring))

	return (unsafe.Pointer)(QSplashScreen_virtualbase_Metacast(unsafe.Pointer(this.h), param1_Cstring))
}

func (this *QSplashScreen) OnMetacast(slot func(super func(param1 string) unsafe.Pointer, param1 string) unsafe.Pointer) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QSplashScreen_override_virtual_Metacast(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QSplashScreen_Metacast
func miqt_exec_callback_QSplashScreen_Metacast(self QSplashScreen, cb intptr_t, param1 *const_char) unsafe.Pointer {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(param1 string) unsafe.Pointer, param1 string) unsafe.Pointer)
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}
	// Convert all CABI parameters to Go parameters
	param1_ret := param1
	slotval1 := GoString(param1_ret)
	virtualReturn := gofunc((&QSplashScreen{h: self}).callVirtualBase_Metacast, slotval1)

	return virtualReturn
}
